src/WebJobs.Script.WebHost/Controllers/KeysController.cs

```diff
@@ -60,6 +62,15 @@ public async Task<IActionResult> Get()
         {
             string hostKeyScope = GetHostKeyScopeForRequest();
 
+            if (string.Equals(hostKeyScope, HostKeyScopes.SystemKeys, StringComparison.OrdinalIgnoreCase) &&
+                _hostManager.State != ScriptHostState.Offline)
+            {
+                // Extensions that are webhook providers create their default system keys
+                // as part of host initialization (when those keys aren't already present).
+                // So we must delay key retrieval until host initialization is complete.
+                await _hostManager.DelayUntilHostReady();
```

Comment:
If the host is not initialized yet, this will introduce a slight delay, affecting ARM/Portal calls to these management APIs. That's why I'm pulling the fix for #4095 in as well - we don't want to be blocking until all listeners are started.